packages/eslint-plugin/tests/rules/no-inferrable-types.test.ts

```diff
@@ -156,7 +156,26 @@ class Foo {
 
   invalid: [
     ...invalidTestCases,
-
+    {
+      // This is invalid TS semantic, but it's trivial to make valid anyway
+      code: 'const fn = (a?: number = 5) => {};',
```

Comment:
not sure if we want to handle this in here, but there is one more use case that is broken.

[playground](https://deploy-preview-5342--typescript-eslint.netlify.app/play/#ts=4.7.2&sourceType=module&code=MYewdgzgLgBAZmGBeGAKAhgfgFwzAVwFsAjAUwCdkYBWASmQD4YBvGAXwG4BYAKF+AA26CBBgAVUtBa8YMAB4BCXARIUq1Xm15A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uYDNFpoAQwBGSWkVKoMkftA7RI4MAF8QyoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

```ts
class Test {
  x!: number = 5
}
```

Reply:
I'll add definite assignment to this PR, but the `private` one seems tangential to this.

Reply:
Also, the `abstract y!: number = 5` is too messed up to be autofixed. The initializer needs to be removed as well. It isn't even reported by the rule under the status quo, so I'll just ignore that.